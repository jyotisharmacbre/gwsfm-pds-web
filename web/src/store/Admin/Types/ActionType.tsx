export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
