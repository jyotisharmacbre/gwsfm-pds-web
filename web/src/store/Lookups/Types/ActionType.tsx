export enum ActionType {
  LOOKUP_PROJECT_STATUS_GET_SUCCESS = 'LOOKUP_PROJECT_STATUS_GET_SUCCESS',
  LOOKUP_PROJECT_STATUS_GET_ERROR = 'LOOKUP_PROJECT_STATUS_GET_ERROR'
}
