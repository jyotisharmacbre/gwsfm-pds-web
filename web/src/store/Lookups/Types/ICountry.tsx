export interface ICountry {
    countryId: number,
    name: string,
    code: string,
    isoAlpha2Code: string,
    currencyId: number
}
