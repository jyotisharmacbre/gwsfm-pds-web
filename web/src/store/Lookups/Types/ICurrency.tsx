export interface ICurrency {
  currencyId: number;
  currencyName: string;
  currencySymbol: string;
  isActive: boolean;
}
