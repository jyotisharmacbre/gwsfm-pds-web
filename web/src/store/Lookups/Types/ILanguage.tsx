
export interface ILanguage {  
  languageID: number,
  name: string,
  shortName: string
}
