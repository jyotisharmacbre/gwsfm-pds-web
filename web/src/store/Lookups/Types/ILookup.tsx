export interface ILookup {
  lookupId: number;
  lookupItem: string;
  lookupKey: number;
  description: string;
}
