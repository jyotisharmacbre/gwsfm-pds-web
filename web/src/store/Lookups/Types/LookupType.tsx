export enum LookupType {
  Currency = 'Currency',
  Project_Status = 'Project_Status'
}
