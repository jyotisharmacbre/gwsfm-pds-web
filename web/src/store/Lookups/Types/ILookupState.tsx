import { ILookup } from './ILookup';
import { ICurrency } from './ICurrency';
import { ILanguage } from './ILanguage';
import { ICountry } from './ICountry';

export interface ILookupState {
  projectstatus: Array<ILookup>;
  currencies: Array<ICurrency> | null;
  languages: Array<ILanguage> | null;
  countries: Array<ICountry> | null;
  error: string | null;
}
