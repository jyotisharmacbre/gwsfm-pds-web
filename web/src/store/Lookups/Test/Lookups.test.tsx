
import { ActionType } from '../../../store/Lookups/Types/ActionType';
import { ILookupState } from '../Types/ILookupState';
import lookupReducer from '../Reducer';
import { currencies, countries, projectstatus, languages } from './lookupData';

const initialState: ILookupState = {
  projectstatus,
  currencies,
  languages,
  countries,
  error: null
};

describe('Lookup reducer', () => {
  it('should handle default action', () => {
    const defaultAction: any = {
      type: ActionType.DEFAULT
    };
    expect(
      lookupReducer(initialState, defaultAction)
    ).toMatchSnapshot();
  });

  it('should handle LOOKUP_PROJECT_STATUS_GET_SUCCESS successfully', () => {
    const getProjectStatusAction: any = {
      type: ActionType.LOOKUP_PROJECT_STATUS_GET_SUCCESS,
      payload: [{
        lookupId: 1,
        lookupItem: 'Project_Status',
        lookupKey: 1,
        description: 'Initial Customer Inquiry'
      }]
    };
    expect(
      lookupReducer(initialState, getProjectStatusAction)
    ).toMatchSnapshot();
  });

  it('should handle getProjectStatus error', () => {
    const getProjectStatusErrorAction: any = {
      type: ActionType.LOOKUP_PROJECT_STATUS_GET_ERROR,
      error: { success: false }
    };
    expect(
      lookupReducer(initialState, getProjectStatusErrorAction)
    ).toMatchSnapshot();
  });

  it('should handle GET_ALL_CURRENCIES_SUCCESS', () => {
    const getCurrenciesSuccessAction: any = {
      type: ActionType.GET_ALL_CURRENCIES_SUCCESS,
      payload: currencies
    };
    expect(
      lookupReducer(initialState, getCurrenciesSuccessAction)
    ).toMatchSnapshot();
  });

  it('should handle GET_ALL_CURRENCIES_ERROR', () => {
    const getCurrenciesErrorAction: any = {
      type: ActionType.GET_ALL_CURRENCIES_ERROR,
      error: { success: false }
    };
    expect(
      lookupReducer(initialState, getCurrenciesErrorAction)
    ).toMatchSnapshot();
  });

  it('should handle GET_ALL_COUNTRIES_SUCCESS', () => {
    const getCountriesSuccessAction: any = {
      type: ActionType.GET_ALL_COUNTRIES_SUCCESS,
      payload: countries
    };
    expect(
      lookupReducer(initialState, getCountriesSuccessAction)
    ).toMatchSnapshot();
  });

  it('should handle GET_ALL_COUNTRIES_ERROR', () => {
    const getCountriesErrorAction: any = {
      type: ActionType.GET_ALL_COUNTRIES_ERROR,
      error: { success: false }
    };
    expect(
      lookupReducer(initialState, getCountriesErrorAction)
    ).toMatchSnapshot();
  });
});