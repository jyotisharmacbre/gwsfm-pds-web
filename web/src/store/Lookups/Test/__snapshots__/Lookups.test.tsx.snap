// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lookup reducer should handle GET_ALL_COUNTRIES_ERROR 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": undefined,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle GET_ALL_COUNTRIES_SUCCESS 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": null,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle GET_ALL_CURRENCIES_ERROR 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": undefined,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle GET_ALL_CURRENCIES_SUCCESS 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": null,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle GET_ALL_LANGUAGES_ERROR 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": undefined,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle GET_ALL_LANGUAGES_SUCCESS 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": null,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle LOOKUP_PROJECT_STATUS_GET_SUCCESS successfully 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": null,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle default action 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": null,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;

exports[`Lookup reducer should handle getProjectStatus error 1`] = `
Object {
  "countries": Array [
    Object {
      "code": "AFG",
      "countryId": 1,
      "isoAlpha2Code": "AF",
      "name": "fghanistan",
    },
  ],
  "currencies": Array [
    Object {
      "currencyId": 4,
      "currencyName": "AED",
      "currencySymbol": "",
    },
  ],
  "error": undefined,
  "languages": Array [
    Object {
      "languageID": 1,
      "name": "English",
      "shortName": "en",
    },
  ],
  "projectstatus": Array [
    Object {
      "description": "Initial Customer Inquiry",
      "lookupId": 1,
      "lookupItem": "Project_Status",
      "lookupKey": 1,
    },
  ],
}
`;
